### PR TITLE
Use Discovery API for Wallets

### PR DESCRIPTION
**Pull request by gregsantos**
### Use Discovery API to fetch and display available wallets

This PR changes hard coding of wallets to use the Discovery API.

- Fetch available wallets from discovery on mount of WalletModal
- Display details based on service data

**TODO**
Future updates to improve
- Once Dapper Wallet is added to Discovery, hard coding of details can be removed and replaced with API response
- Lilico update will allow for opening directly vs using current Discovery iFrame strategy
- `configureFCLAndLogin` can be replaced with direct calls to `fcl.authenticate({ service })` once above criteria are met
- Wallet logos should be URI based in service defs from Discovery API

### CODE REVIEW

**Comment by gregsantos** [comment]
Any suggestions or updates required for this to be reviewed/merged?

File: src/lib/flow/actions.js [hunk 1]
```diff
@@ -69,14 +69,14 @@ const configureFCL = (wallet) => {
...
       .put("discovery.wallet", import.meta.env.VITE_FCL_DISCOVERY)
       .put("discovery.wallet.method", "IFRAME/RPC")
   }
 }
 
-export const configureFCLAndLogin = async (wallet) => {
+export const configureFCLAndLogin = async (wallet, service) => {
   currentWallet.set(wallet);
   configureFCL(wallet);
   await fcl.authenticate();
```
Comment (muttoni) [inline src/lib/flow/actions.js:82]: I think this line is missing something, it throws an error. 

image

I assume it needs to be changed to: `fcl.authenticate({service: service})`. that makes it it works for Lilico. 

However Blocto still doesn't work, for another reason (throws 404). Any ideas?

image

File: src/lib/components/WalletModal.svelte [hunk 2]
```diff
@@ -12,24 +31,20 @@
...
       <span>Connect Dapper</span>
     </button>
   </div>
-  <div class="wallet">
-    <button id="other" on:click={() => configureFCLAndLogin("other")}>
-      <img src="/lilico-logo.jpg" alt="lilico logo" />
-      <span>Connect Lilico</span>
-    </button>
-  </div>
+  {#each services as service}
```
Comment (muttoni) [inline src/lib/components/WalletModal.svelte:40]: This is much more elegant than hardcoding wallets, BUT, it causes the issue of showing only Dapper wallet for 2-3 seconds before the rest of the wallets show up. I suggest we either preload all wallets on page load, OR show loading indicator for ALL wallets until services are loaded.

**Review by muttoni — CHANGES_REQUESTED** [review]
Thanks Greg, sorry for delay here. Looking forward to merging this, just found a couple things  (see review comments)

File: src/lib/components/WalletModal.svelte (hunk 2 above)
Comment (gregsantos) [reply src/lib/components/WalletModal.svelte:40]: Per favore scusi! Il mio Svelte non è molto buono!
Battled that buffer issue again today. Think we'll need to sync unless you had any luck upgrading FCL to latest

**Pull request closed without merging**